lms/djangoapps/instructor/utils.py

```diff
@@ -185,16 +183,32 @@ def collect_ora2_data(course_id):
 
 
 # pylint: disable=invalid-name
-def ora2_data_queries():
+def ora2_data_queries(is_anonymous):
     """
     Wraps a raw SQL query which retrieves all ORA2 responses for a course.
     """
-
-    RAW_QUERY = """
+    if is_anonymous:
+        RAW_QUERY_3 = """
+        `student`.`student_id` AS `anonymized_student_id`,
+        """
+    else:
+        RAW_QUERY_3 = """
+        (
+            SELECT `auth_user`.`email`
+            FROM `auth_user`
+            JOIN `student_anonymoususerid` AS `anonymous`
+            ON `auth_user`.`id` = `anonymous`.`user_id`
+            WHERE `student`.`student_id` = `anonymous`.`anonymous_user_id`
+        ) AS `email`,
+        """
+
+    RAW_QUERY_1 = """
 SET SESSION group_concat_max_len = 1000000;
 SELECT `sub`.`uuid` AS `submission_uuid`,
 `student`.`item_id` AS `item_id`,
-`student`.`student_id` AS `anonymized_student_id`,
+"""
```

Comment:
Nit: If this line stays in the method, the triple quotes should be aligned with RAW_QUERY_1.